HOTT-1076 Show commodity code in 4 4 2 pattern

jebw opened the pull request:
### Jira link

[HOTT-1076](https://transformuk.atlassian.net/browse/HOTT-1076)

### What?

I have added/removed/altered:

- [x] Added a helper for format the commodity code into a 4 4 2 split, this will also handle heading or chapter codes if handed those
- [x] Changed the _search partial to use the helper when formatting the heading

### Why?

I am doing this because:

- We want to visually break up the commodity code

### Have you? (optional)

- [x] Reviewed view changes with stake holders
- [x] Validated mobile responsive behaviour of view changes

### Deployment risks (optional)

- None

### Notes

This conflicts with the refactor of the `_search` partial in #393 - it should be rebased and merged after that PR.

### Before

![Screenshot from 2021-11-10 11-10-49](https://user-images.githubusercontent.com/10818/141103438-ba710ba4-73d8-4aab-af5f-abbcc9dfe310.png)

### After

![Screenshot from 2021-11-10 11-10-10](https://user-images.githubusercontent.com/10818/141103458-7318c657-5ad4-4268-9b89-c2b5f5b8a50e.png)

jebw committed:
HOTT-1076 Show commodity code in 4 4 2 pattern

alexdesi commented on app/helpers/commodities_helper.rb:
```diff
@@ -144,4 +144,14 @@ def declarable_heading_full(commodity)
     end
   end
+
+  def four_four_two_commodity_code(code)
```
the name is a bit cryptic, what about 
`pretty_commodity_code` or `readable_commodity_code`?

jebw replied:
Yeh - naturally I'd call it something like `format_commodity_code` but we're already formatting it several different ways - hence trying to incorporate the format into the name.

How about `segmented_commodity_code` ?

jebw replied:
Renamed to `segmented_commodity_code`

alexdesi commented on app/helpers/commodities_helper.rb:
```diff
@@ -144,4 +144,14 @@ def declarable_heading_full(commodity)
+
+    parts = code.to_s.gsub(/[^\d]/, '').split('').each_slice(4).map(&:join)
+
+    tag.span class: 'commodity-code-4-4-2' do
```
I think this should be renamed too: the `4-2-2` part resembles some kind of formatting we are gonna use.

jebw replied:
renamed to `.segmented-commodity-code`

alexdesi commented on app/views/shared/_search.html.erb:
```diff
@@ -2,7 +2,7 @@
     <header>
       <h1 class="govuk-heading-l commodity-header" data-comm-code="<%= commodity_code %>">
-        Commodity information for <%= commodity_code %>
+        Commodity information for <%= four_four_two_commodity_code commodity_code %>
```
this is not the only place to apply prettify the code, is it?

jebw replied:
Ticket requests we do it just in the heading first - then it will be rolled out elsewhere later?

alexdesi reviewed (approved):
Nice one!
I am just not sure about the naming of the method.

jebw committed:
HOTT-1076 Renamed comm code helper

From `four_four_two_commodity_code` to `segmented_commodity_code` - addressing review comments

jebw committed:
HOTT-1076 Shortened page heading

jebw committed:
HOTT-1076 Fixed comm code copy-and-paste behaviour

Now you get the unsegmented version with both the copy link and by highlighting and ctrl+c

alexdesi reviewed (approved):
It looks good to me!
